src/mscorlib/src/System/Diagnostics/Eventing/XplatEventLogger.cs

```diff
@@ -109,24 +109,53 @@ private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnly
                 sb.Append("\\\"");
                 sb.Append(':');
 
-                var valuestr = payload[i] as string;
-
-                if( valuestr != null)
-                {
-                    sb.Append("\\\"");
-                    minimalJsonserializer(valuestr,sb);
-                    sb.Append("\\\"");
-                }
-                else
+                switch(payload[i])
                 {
-                    sb.Append(payload[i].ToString());
+                    case string str:
+                    {
+                        sb.Append("\\\"");
+                        minimalJsonserializer(str, sb);
+                        sb.Append("\\\"");
+                        break;
+                    }
+                    case byte[] byteArr:
+                    {
+                        sb.Append("\\\"");
+                        AppendByteArrayAsHexString(sb, byteArr);
+                        sb.Append("\\\"");
+                        break;
+                    }
+                    default:
+                    {
+                        sb.Append(payload[i].ToString());
+                        break;
+                    }
                 }
-
             }
             sb.Append('}');
             return StringBuilderCache.GetStringAndRelease(sb);
         }
 
+        private static void AppendByteArrayAsHexString(StringBuilder builder, byte[] byteArray)
+        {
+            Debug.Assert(builder != null);
```

Comment:
nit: it's impossible for the builder to be null here (the calling method is using it)

Reply:
You're right.  This just protects the method against future changes.